pkgs/misc/vscode-extensions/vscodeWithConfiguration.nix

```diff
@@ -0,0 +1,39 @@
+# wrapper over vscode to control extensions per project (extensions folder will be created in execution path)
+{ pkgs                            ? import <nixpkgs> {}
+, lib                             ? pkgs.lib
+, writeShellScriptBin             ? pkgs.writeShellScriptBin
+, extensionsFromVscodeMarketplace ? pkgs.vscode-utils.extensionsFromVscodeMarketplace
+
+##User input
+
+, vscode                          ? pkgs.vscode
+# extensions to be symlinked into the project's extensions folder
+, nixExtensions                   ? []
+# extensions to be copied into the project's extensions folder
+, mutableExtensions               ? []
+, vscodeExtsFolderName            ? ".vscode-exts"
+}:
+let 
+  nixExtsDrvs = extensionsFromVscodeMarketplace nixExtensions;
+  mutExtsDrvs = extensionsFromVscodeMarketplace mutableExtensions;
+
+  #removed not defined extensions
+  rmExtensions =  lib.optionalString (nixExtensions++mutableExtensions != []) ''
+    find ${vscodeExtsFolderName} -mindepth 1 -maxdepth 1 ${lib.concatMapStringsSep " " (e : ''! -iname ${e.publisher}.${e.name}'') (nixExtensions++mutableExtensions)} -exec sudo rm -rf {} \;
+  '';
+  #copy mutable extension out of the nix store
+  cpExtensions = ''
+    ${lib.concatMapStringsSep "\n" (e : ''ln -sfn ${e}/share/vscode/extensions/* ${vscodeExtsFolderName}/'') nixExtsDrvs}
+    ${lib.concatMapStringsSep "\n" (e : ''
+      cp -a ${e}/share/vscode/extensions/${e.vscodeExtUniqueId} ${vscodeExtsFolderName}/${e.vscodeExtUniqueId}-${(lib.findSingle (ext: ''${ext.publisher}.${ext.name}'' == e.vscodeExtUniqueId) "" "m" mutableExtensions ).version}
```

Comment:
`cp -a` keeps the directory permissions the same, so the extension directory wouldn't be mutable no? The extension directory permissions for me are 555.

Reply:
for me on drawin it solved the permission problem on the csharp extension. but maybe it is not the same on a different OS. will changing permissions make sense?